tests/csapi/room_typing_test.go

```diff
@@ -40,3 +40,55 @@ func TestTyping(t *testing.T) {
 		return false
 	}))
 }
+
+// sytest: Typing notifications don't leak
+func TestLeakyTyping(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintOneToOneRoom)
+	defer deployment.Destroy(t)
+
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	bob := deployment.Client(t, "hs1", "@bob:hs1")
+	charlie := deployment.RegisterUser(t, "hs1", "charlie", "charliepassword", false)
+
+	roomID := alice.CreateRoom(t, map[string]interface{}{"preset": "public_chat"})
+	bob.JoinRoom(t, roomID, nil)
+
+	bobToken := bob.MustSyncUntil(t, client.SyncReq{}, client.SyncJoinedTo(bob.UserID, roomID))
+
+	_, charlieToken := charlie.MustSync(t, client.SyncReq{TimeoutMillis: "0"})
+
+	alice.MustDoFunc(t, "PUT", []string{"_matrix", "client", "v3", "rooms", roomID, "typing", alice.UserID}, client.WithJSONBody(t, map[string]interface{}{
+		"typing":  true,
+		"timeout": 10000,
+	}))
+
+	bob.MustSyncUntil(t, client.SyncReq{Since: bobToken}, client.SyncEphemeralHas(roomID, func(result gjson.Result) bool {
+		if result.Get("type").Str != "m.typing" {
+			return false
+		}
+		for _, item := range result.Get("content").Get("user_ids").Array() {
+			if item.Str == alice.UserID {
+				return true
+			}
+		}
+		return false
```

Comment:
This checks that alice is in the list of typing users. Can we change this to check that `[alice]` is exactly the list of typing users, e.g. with `JSONCheckOff`? (Which, in fairness, the original sytest didn't do)

You might also prefer to `.Get("content.user_ids"`). I'm not sure what the preferred idiom is.

Reply:
Correct on both counts, `JSONCheckOff` could also give a stronger guarantee that there are no additional users in the typing list as well, thanks for the suggestions 👍

Reply:
`must` can't work with raw GJson objects yet, so using JSONCheckOff specifically wont be possible: https://github.com/matrix-org/complement/issues/528